### PR TITLE
fix(chat): fix application edit, fix application delete & icon fallback logic (Issue #2230 #2021)

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

- fallback to defaultl icon if there is no icon url
- keep installed applications ids as reference
- select unavailable model and pass isModelUnavailable flag instead of unavailableModelId in chat settings

Issues:

- Issue #2068
- Issue #2230 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2245.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ConversationSettings.tsx [hunk 1]
```diff
@@ -183,10 +183,13 @@ export const ConversationSettings = ({
         <div className="shrink bg-layer-2 px-3 py-4 md:px-6">
           <ConversationSettingsModel
             conversation={conversation}
-            modelId={model?.reference ?? recentModelsIds[0]}
-            unavailableModelId={
-              !model?.id && !isPseudoModel(modelId) ? modelId : undefined
+            modelId={
+              model?.reference ??
+              (!model?.id && !isPseudoModel(modelId)
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ConversationSettings.tsx]: extra check here:
```suggestion
              (!isPseudoModel(modelId)
```

File: apps/chat/src/components/Chat/ConversationSettings.tsx [hunk 2]
```diff
@@ -183,10 +183,13 @@ export const ConversationSettings = ({
...
+                ? modelId
+                : recentModelsIds[0])
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ConversationSettings.tsx]: it looks like `recentModelsIds[0]` will be used only if model is undefined and it's pseudomodel... it's strange

File: apps/chat/src/store/models/models.epics.ts [hunk 3]
```diff
@@ -153,7 +153,7 @@ const getInstalledModelIdsEpic: AppEpic = (action$, state$) =>
                 getRootId({ featureType: FeatureType.Application }),
               ),
             )
-            .map((app) => app.id);
+            .map((app) => app.reference);
           const installedModelIds = new Set(
             installedModels.map((model) => model.id),
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/models/models.epics.ts:158]: maybe lets try to use references here too? (157 line)

Comment (IlyaBondar) [reply apps/chat/src/store/models/models.epics.ts:158]: @Alexander-Kezik , good catch

Comment (Gimir) [reply apps/chat/src/store/models/models.epics.ts:158]: there is no reference, it is installedModels array, which have { id: string; pinned: boolean; } type. It is already references.

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2245.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**